deeprankcore/dataset.py

```diff
@@ -234,12 +242,13 @@ def load_one_graph(self, fname, mol): # noqa
                 y = None
             else:
                 if targets.VALUES in grp and self.target in grp[targets.VALUES]:
-                    try:
-                        y = torch.tensor([grp[f"{targets.VALUES}/{self.target}"][()]], dtype=torch.float).contiguous().to(self.device)
-                    except Exception as e:
-                        _log.error(e)
-                        _log.info('If your target variable contains categorical classes, \
-                        please convert them into class indices before defining the GraphDataset instance.')
+                    y = torch.tensor([grp[f"{targets.VALUES}/{self.target}"][()]], dtype=torch.float).contiguous().to(self.device)
+
+                    if self.task == targets.REGRESS and self.target_transform is True:
+                        y = torch.sigmoid(torch.log(y))
+                    elif self.task is not targets.REGRESS and self.target_transform is True:
+                        raise ValueError(f"Task is set to {self.task}. Please set it to regress to transform the target with a sigmoid.")
```

Comment:
Is it necessary to raise an error in this case? Maybe a warning/log input is sufficient and then `target_transform` is just ignored.

Reply:
Not necessary, but conceptually is wrong. Also, I'll change this again with the transformations PR, which will include also target transformations. In case we can change this there :)